share/sysbuild/Kconfig.image.template

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2023 Nordic Semiconductor
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# This is a Kconfig @template@ file used for passing sysbuild controlled Kconfig
+# settings from sysbuild to images.
+
+config SYSBUILD_CONTROLLED
+	bool
+	default y
+	help
+	  Setting that can be used to indicate this setting is controlled by
+	  sysbuild when sysbuild is used as high level build system.
+
+config SYSBUILD_BOARD
+	string
+	default "@BOARD@"
+	help
+	  High-level board defined in sysbuild.
+
+if BOARD = SYSBUILD_BOARD
+
+config BOOTLOADER_MCUBOOT
+	default y if "@SB_CONFIG_BOOTLOADER_MCUBOOT@" && !MCUBOOT
+
+if BOOTLOADER_MCUBOOT
+
+config MCUBOOT_SIGNATURE_KEY_FILE
+	default "@SB_CONFIG_BOOT_SIGNATURE_KEY_FILE@"
+
+config MCUBOOT_GENERATE_UNSIGNED_IMAGE
+	default y if "@SB_CONFIG_SIGNATURE_TYPE@" = "NONE"
```

Comment:
I'm yet to get the PR and try it, but what would happen if someone configured a project with this set to none, then changed it in `ninja sysbuild_menuconfig` to a different mode, would the new option propagate to the images or would the old one persist (i.e. without pristine)?

Reply:
> would the new option propagate to the images or would the old one persist (i.e. without pristine)?

in this case, `MCUBOOT_GENERATE_UNSIGNED_IMAGE`, then it would not be persistent, because the prompt of the original symbol is depending on `!SYSBUILD_CONTROLLED`.
So when sysbuild is used then the symbol is promptless and thus the default will always be the used value.
The default is controlled by sysbuild and regenerated when `sysbuild_menuconfig` has changed any setting in the sysbuild `.config`.
Ref: https://github.com/zephyrproject-rtos/zephyr/blob/0da5e417014a86b5cb220e3be2b84c4389ddff1d/Kconfig.zephyr#L870

But there is a risk, for example if a oot module redefines the setting and also gives it a prompt, but don't set the `if !SYSBUILD_CONTROLLED`, then the downstream setting can cause the setting to become `sticky`.

Similar with choices, where a downstream may extend a choice in Zephyr with extra entries, like it's done here:
https://github.com/nrfconnect/sdk-nrf/blob/6c606e116b8ef0f015a9e256ee557118f2d33f7b/Kconfig.nrf#L27
where a downstream project extends the Zephyr choice from here:
https://github.com/zephyrproject-rtos/zephyr/blob/68955ebdb6d8a9c6598810446fecc316209089e1/modules/mbedtls/Kconfig#L24

in Zephyr, we can have the choice in Zephyr ensuring only the sysbuild selected entry is available (and thereby the choice selection will follow the sysbuild selection), but in case an oot module provides extra choice entries, then the oot module must ensure to follow same principle, else the selection has a risk of becoming sticky.

So the principle in this PR does has it's limitation.
The good thing in the principle presented with this PR, is that the settings becomes promptless when sysbuild is in use, and thereby users will not be mistake reconfigure the setting directly in the sample.

And alternative approach can be to have sysbuild generating a config fragment, and then use this fragment to forcefully update the image `.config` and thereby overruling.
A benefit on such solution can be that sysbuild always win.
Disadvantage is that the setting is available in the image's menuconfig for configuration and users may fail to understand why their new selection / value is overruled / ignored.
Also it might require some more tweaking around Zephyr's kconfiglib usage.

Reply:
yes, cause the MCUboot Kconfig which specifies the choice has not been adjusted yet in this PR.
Needs to be adjusted according to description in above comment, wrt. `choice`.

before pushing additional changes to Zephyr modules, then I like to get feedback on the direction / principle first. Note, this is another reason I marked this PR as `draft`.

All settings defined in Zephyr repo Kconfig should behave correctly in this PR.

Reply:
You may say that this PR introduces the infrastructure / possibility of controlling Kconfigs from sysbuild, but doesn't update Kconfigs in Zephyr modules to take advantage of this infrastructure.